tests/lib/rules/prefer-destructuring.js

```diff
@@ -139,27 +139,39 @@ ruleTester.run("prefer-destructuring", rule, {
     invalid: [
         {
             code: "var foo = array[0];",
+            output: null,
```

Comment:
Could we add a test for nested destructuring just to document what the expected behavior of the autofixer is here?

Reply:
The test included this case where the code could be fixed into nested destructuring, but is fixed into the unnested form.

https://github.com/eslint/eslint/blob/55f1ae6de889d74a4d8c7074024726576ba5e085/tests/lib/rules/prefer-destructuring.js#L164-L171

Do I need to add `var {foo: {bar}} = object` as a valid case?

Reply:
> Do I need to add `var {foo: {bar}} = object` as a valid case?

sound good if it has not been included. thanks!